sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -471,14 +471,14 @@ def __init__(self, you_should_set_this_=None):
 
 def test_check_estimator_pairwise():
     # check that check_estimator() works on estimator with _pairwise
-    # kernel or  metric
+    # kernel or metric
 
     # test precomputed kernel
     est = SVC(kernel='precomputed')
     check_estimator(est)
 
     # test precomputed metric
-    est = KNeighborsRegressor(metric='precomputed')
+    est = KNeighborsRegressor()
```

Comment:
You should not remove `metric='precomputed'`, which is essential to this test.
Why is `test_check_estimator_pairwise` related to your new multioutput tests?

Reply:
Hey @TomDLT, thanks for the super fast response! :)

If I set `metric='precomputed'` I get `ValueError: Negative values in data passed`.

Originally I fixed it by adding `RequiresPositiveDataMixin` to the `KNeighborsRegressor` and then by removing `metric='precomputed'`. After reading deeper I agree neither is a good idea.

How about only setting the `requires_positive_data` tag if `self.metric == 'precomputed'`? It feels kinda hacky but solves it. See latest commit.

Reply:
Oh I see, it comes from the `predict` in `check_regressor_multioutput`, which checks that precomputed distances are non-negative, whereas the `fit` does not check (but probably should).

I don't see any obvious reason to protest against your conditional tag:
```py
    def _more_tags(self):
        if self.metric == 'precomputed':
            return {'requires_positive_data': True}
        else:
            return {}
```
but I would prefer to have some input from other people involved in estimator tags.
@amueller @jnothman @rth @glemaitre.

Reply:
Any further comments @amueller @jnothman @rth @glemaitre @TomDLT or can we merge?